starlite/asgi/routing_trie/validate.py

```diff
@@ -0,0 +1,29 @@
+from typing import TYPE_CHECKING
+
+from starlite.asgi.routing_trie.types import PathParameterSentinel
+from starlite.exceptions import ImproperlyConfiguredException
+
+if TYPE_CHECKING:
+    from starlite.asgi.routing_trie.types import RouteTrieNode
+
+
+def validate_node(node: "RouteTrieNode") -> None:
+    """Recursively traverses the trie from the given node upwards.
+
+    Args:
+        node: A trie node.
+
+    Raises:
+        ImproperlyConfiguredException
+
+    Returns:
+        None
+    """
+    if node["is_asgi"] and bool(set(node["asgi_handlers"]).difference({"asgi"})):
```

Comment:
This could be simplified: 

```suggestion
    if node["is_asgi"] and bool({"asgi"}.difference(node["asgi_handlers"])):
```